src/typings.ts

```diff
@@ -177,32 +206,71 @@ function createExportedTypes(
   if (propTypes || classComponent) {
     m.members.push(interf);
   }
-
   if (classComponent) {
-    if (!exportType) {
```

Comment:
This function exceeded the `cyclomatic-complexity` linting rule (~7 if statements), so I had to break it down further (a function for classes and a function for functional components). 

Side note - There's a lot of passing around of args in this file, could maybe refactor them into a single object and/or move these functions inside `createTypings`.

Reply:
If you like you can refactor the args passing, but I would request to do that in a different PR.
Otherwise this will grow bigger and bigger.